Lib/logging/__init__.py

```diff
@@ -413,38 +416,54 @@ def makeLogRecord(dict):
     rv.__dict__.update(dict)
     return rv
 
+
 #---------------------------------------------------------------------------
 #   Formatter classes and functions
 #---------------------------------------------------------------------------
-
 class PercentStyle(object):
 
     default_format = '%(message)s'
     asctime_format = '%(asctime)s'
     asctime_search = '%(asctime)'
+    validation_pattern = r'(\%\([^%()]+\)[#|0|\-|\s|+]?[\d]*[d|i|o|u|x|X|e|E|f|E|g|G|C|r|s|a|%]{1})'
```

Comment:
This still has some problems:
* It is still not correct with respect to the spec - for example it doesn't handle the use of literal `*` to specify width and precision.
* It's needlessly verbose - `[d|i|o|u|x|X|e|E|f|E|g|G|C|r|s|a|%]{1}` could be replaced with just `[diouxefgcrsa]` if compiled with `re.I`- excluding the `%` character, as that wouldn't make sense as part of a `%(name)` format specifier, and the `{1}` is redundant.
* The bit between the parentheses - why not just use `\w+`?
* When specifying character alternatives, you don't need a `|` - `[#0+ -]` would be preferable to `[#|0|\-|\s|+]`, leaving the `-` at the end to avoid ambiguity, as otherwise `-` inside `[...]` has special meaning.
* You didn't take up my point about using a compiled pattern. Did you not understand what I meant by that, or did you just forget it?

I suggest you just use `re.compile(r'%\(\w+\)[#0+ -]?(\*|\d+)?(\.(\*|\d+))?[diouxefgcrsa]', re.I)` as the `validation_pattern`.